scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java

```diff
@@ -1162,47 +1155,81 @@ public boolean equals(Object obj) {
         return false;
     }
 
+    public synchronized void updateVariables(SchedulingService schedulingService) throws IOException, ClassNotFoundException {
+        if (updatedVariables == null) {
+            updatedVariables = new LinkedHashMap<>();
+            updatedVariables.putAll(internalJob.getVariables());
+            for (TaskVariable variable : variables.values()) {
+                if (!variable.isJobInherited()) {
+                    updatedVariables.put(variable.getName(), variable.getValue());
+                }
+            }
+
+            if (ideps != null) {
+                List<TaskId> parentIds = new ArrayList<>(ideps.size());
+                for (InternalTask parentTask : ideps) {
+                    parentIds.add(parentTask.getId());
+                }
+                if (parentIds.size() > 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1155)